src/bin/nydus-image/main.rs

```diff
@@ -989,6 +996,22 @@ impl Command {
         Ok(blob_id)
     }
 
+    fn get_blob_size(matches: &clap::ArgMatches, ty: ConversionType) -> Result<u64> {
+        if ty != ConversionType::StargzIndexToRef {
+            return Ok(0);
+        }
+
+        match matches.value_of("blob-data-size") {
+            None => bail!("no value specified for '--blob-data-size'"),
```

Comment:
Does it seem must be calculated by `blob_size - footer_size` in snapshotter?

Reply:
It's the toc offset from estargz image. When snapshotter tries to downloaded the TOC, it already gets the `blob-data-size` by parsing the eStargz footer.